workspaces/bi/bi-extension/src/test/e2e-playwright-tests/api-services/graphqlUtils.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Frame } from '@playwright/test';
+import { page } from '../utils';
+
+// Centralized test data for all GraphQL service E2E tests
+export const TEST_DATA = {
+    service: {
+        basePath: (attempt: number) => `/sample${attempt}`,
```

Comment:
We'll keep graphql as the initial base path, since that's the expected default behavior.